widgets/text/src/widget.tsx

```diff
@@ -1,73 +1,90 @@
-// @ts-nocheck File not migrated fully to TS
 export {};
 
+const t = Stage.Utils.getT('widgets.text');
+
+type Configuration = {
+    header: string;
+    content: string;
+    headerTextFont: string;
+    headerTextColor: string;
+    headerTextSize: number;
+    contentTextFont: string;
+    contentTextColor: string;
+    contentTextSize: number;
+};
+
 const textFonts = [
     { name: 'Sans Serif', value: 'sans-serif' }, // used as default
     { name: 'Serif', value: 'serif' },
     { name: 'Monospace', value: 'monospace' }
 ];
 
-Stage.defineWidget({
+Stage.defineWidget<unknown, unknown, Configuration>({
     id: 'text',
-    name: 'Text box',
-    description: 'Adds text box',
+    name: t('name'),
+    description: t('description'),
     showHeader: false,
     showBorder: false,
     isReact: true,
     hasReadme: true,
     permission: Stage.GenericConfig.WIDGET_PERMISSION('text'),
     initialConfiguration: [
-        { id: 'header', name: 'Header', default: 'Header text', type: Stage.Basic.GenericField.STRING_TYPE },
+        {
+            id: 'header',
+            name: t('configuration.header.name'),
+            default: t('configuration.header.default'),
+            type: Stage.Basic.GenericField.STRING_TYPE
+        },
         {
             id: 'content',
-            name: 'Content',
-            description: 'Markdown is supported',
-            default: 'Markdown supported content. Update in widget configuration.',
+            name: t('configuration.content.name'),
+            description: t('configuration.description.name'),
+            default: t('configuration.content.default'),
             type: Stage.Basic.GenericField.CUSTOM_TYPE,
-            component: Stage.Basic.Form.TextArea
+            component: Stage.Basic.Form.TextArea as Stage.Types.WidgetComponent
         },
 
         {
             id: 'headerTextColor',
-            name: 'Header text color',
+            name: t('configuration.headerTextColor.name'),
             default: '#000000',
             type: Stage.Basic.GenericField.CUSTOM_TYPE,
             component: Stage.Basic.Form.ColorPicker
         },
         {
             id: 'headerTextSize',
-            name: 'Header text size [px]',
+            name: t('configuration.headerTextSize.name'),
             default: '20',
```

Comment:
Hmm... It looks like a bug. Shouldn't it be a number and not a string? 🤔 
```suggestion
            default: 20,
```

The same about `contentTextSize`.

Maybe `StageAPI` should be improved so that `default` property have the same type as value?

Please check if providing number works here and if it works, let's fix it at once.

Reply:
I changed `headerTextSize` and `contentTextSize` to `number` type.

I tested changing numbers in the widget settings and it works.

Reply:
I changed the type of `default` to the same as in `type` in `StageWidgetConfigurationDefinition ` in StageAPI
[7fdf7ef](https://github.com/cloudify-cosmo/cloudify-stage/pull/2216/commits/7fdf7ef7536167a680e04d51416f3dfb1200ff83)

Reply:
```
interface StageWidgetConfigurationDefinition {
    id: string;
     name?: string;
     description?: ReactNode;
     type: GenericFieldType;
(-) default?: any;
(+) default?: GenericFieldType;
     placeHolder?: string;
     hidden?: boolean;
     component?: StageWidgetComponent;
    /** Used for lists */
    items?: (string | { name: string; value: string })[];
    [key: string]: any;
}
```


causing the`check-types` to fail:

```
[2022-10-19T15:24:53.330Z] app/utils/GenericConfig.ts(10,13): error TS2322: Type 'number' is not assignable to type 'GenericFieldType | undefined'.
[2022-10-19T15:24:53.330Z] app/utils/GenericConfig.ts(24,13): error TS2322: Type 'number' is not assignable to type 'GenericFieldType | undefined'.
[2022-10-19T15:24:53.330Z] app/utils/GenericConfig.ts(33,13): error TS2322: Type 'string' is not assignable to type 'GenericFieldType | undefined'.
[2022-10-19T15:24:53.330Z] app/utils/GenericConfig.ts(42,13): error TS2322: Type 'boolean' is not assignable to type 'GenericFieldType | undefined'.
```

I'm working on fixing that.

Reply:
I reverted the commit of changing the type of `default` type. I think maybe the following is the solution:
```
interface StageWidgetConfigurationDefinition<T> {
    id: string;
    name?: string;
    description?: ReactNode;
    type: GenericFieldType;
    default?: T;
    placeHolder?: string;
    hidden?: boolean;
    component?: StageWidgetComponent;
    /** Used for lists */
    items?: (string | { name: string; value: string })[];
    [key: string]: any;
}
```
But I guess it will require changing every widget.

Reply:
Thanks for the efforts. I think that we'll try to fix it within RD-4419, so let's leave it like that.